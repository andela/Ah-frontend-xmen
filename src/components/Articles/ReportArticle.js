import React from 'react';

// eslint-disable-next-line react/prefer-stateless-function

const ReportArticle = props => (
  <div>
    <div className="container edit-container">
      <h6 className="mt-5">Report the Article</h6>
      <form className="mt-2 mb-5" onSubmit={props.onSubmit}>
        <div>
          <textarea name="reason" className="form-control" rows="8" maxLength="360" placeholder="Reason for reporting this article..." onChange={props.onChange} />
        </div>
        <button type="submit" className="btn button-primary mt-3" id="formsubmit">Submit</button>
        <button type="submit" className="btn button-primary mt-3" id="formCancel" onClick={props.onClick}>Cancel</button>
      </form>
    </div>
  </div>
);

export default ReportArticle;
