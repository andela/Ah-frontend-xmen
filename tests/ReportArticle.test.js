import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import ReportArticle from '../src/components/Articles/ReportArticle';

describe('<Report Article />', () => {
  const props = {
    onSubmit: jest.fn(),
    onChange: jest.fn(),
    onClick: jest.fn(),
  };

  it('tests the report Article component matches the snapshot', () => {
    const wrapper = shallow(<ReportArticle {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
