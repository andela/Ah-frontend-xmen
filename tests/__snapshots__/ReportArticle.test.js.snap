// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Report Article /> tests the report Article component matches the snapshot 1`] = `ShallowWrapper {}`;
